x/datapool/keeper/pool_distribute.go

```diff
@@ -0,0 +1,212 @@
+	instantRevenueDistribute.AppendPoolID(poolID)
+	k.SetInstantRevenueDistribute(ctx, instantRevenueDistribute)
+	return nil
+}
+
+func (k Keeper) DistributeRevenuePools(ctx sdk.Context) error {
+	instantRevenueDistribute := k.GetInstantRevenueDistribute(ctx)
+
+	if instantRevenueDistribute.IsEmpty() {
+		return nil
+	}
+
+	// This lastIndex is to execute the following iteration in the next block if it does not end in one block.
+	// But We're currently processing it all in one block. :)
+	lastIndex := 0
+	// TODO We need to think about how to deal with the pool where distribution has failed.
+	for i, poolID := range instantRevenueDistribute.PoolIds {
+		lastIndex = i
+		// search current pool info
+		pool, err := k.GetPool(ctx, poolID)
+		if err != nil {
+			return sdkerrors.Wrap(types.ErrRevenueDistribute, err.Error())
+		}
+
+		// send deposit to curator
+		err = k.sendDepositToCurator(ctx, pool)
+		if err != nil {
+			return err
+		}
+
+		// distribute of revenue to each seller
+		err = k.executeRevenueDistribute(ctx, pool)
+		if err != nil {
+			return err
+		}
+	}
+	// delete this pool from instantRevenueDistribute
+	instantRevenueDistribute.RemovePreviousIndex(lastIndex)
+	k.SetInstantRevenueDistribute(ctx, instantRevenueDistribute)
+
+	return nil
+}
+
+// sendDepositToCurator returns to the Curator if the pool status is ACTIVE but the deposit has not yet been returned.
+func (k Keeper) sendDepositToCurator(ctx sdk.Context, pool *types.Pool) error {
+	if pool.Status != types.ACTIVE || pool.IsPaidDeposit {
+		return nil
+	}
+
+	curatorAddr, err := sdk.AccAddressFromBech32(pool.Curator)
+	if err != nil {
+		return sdkerrors.Wrap(types.ErrRevenueDistribute, err.Error())
+	}
+	poolAddr, err := sdk.AccAddressFromBech32(pool.PoolAddress)
+	if err != nil {
+		return sdkerrors.Wrap(types.ErrRevenueDistribute, err.Error())
+	}
+	err = k.bankKeeper.SendCoins(ctx, poolAddr, curatorAddr, sdk.NewCoins(pool.Deposit))
+	if err != nil {
+		return sdkerrors.Wrap(types.ErrRevenueDistribute, err.Error())
+	}
+	pool.IsPaidDeposit = true
+	k.SetPool(ctx, pool)
+
+	return nil
+}
+
+func (k Keeper) getAvailablePoolCoinAmount(ctx sdk.Context, pool *types.Pool) (*sdk.Int, error) {
+	poolAddr, err := sdk.AccAddressFromBech32(pool.PoolAddress)
+	if err != nil {
+		return nil, sdkerrors.Wrap(types.ErrRevenueDistribute, err.Error())
+	}
+	poolBalance := k.bankKeeper.GetBalance(ctx, poolAddr, assets.MicroMedDenom)
+	if pool.IsPaidDeposit {
+		return &poolBalance.Amount, nil
+	} else {
+		amount := poolBalance.Amount.Sub(pool.Deposit.Amount)
+		return &amount, nil
+	}
+}
+
+func (k Keeper) executeRevenueDistribute(ctx sdk.Context, pool *types.Pool) error {
+	availablePoolCoinAmount, err := k.getAvailablePoolCoinAmount(ctx, pool)
+	if err != nil {
+		return err
+	}
+
+	// calculate the revenue to be sent to each seller(totalSalesBalance / totalShareTokenAmount)
+	eachDistributionAmount := k.getEachDistributionAmount(pool)
```

Comment:
Calculate the revenue to be distributed to each seller.

Reply:
please rename this to `getDistributionPerSeller`.